crates/matrix-sdk/tests/integration/matrix_auth.rs

```diff
@@ -313,3 +320,157 @@ fn test_serialize_session() {
+        })
+        .request_config(RequestConfig::new().disable_retry())
+        .build()
+        .await
+        .unwrap();
+
+    let auth = client.matrix_auth();
+    auth.login_username("example", "hunter2").send().await.unwrap();
+
+    assert!(client.logged_in(), "Client should be logged in");
+    assert!(auth.logged_in(), "Client should be logged in with the MatrixAuth API");
+
+    let me = client.user_id().expect("we are now logged in");
+    let own_identity =
+        client.encryption().get_user_identity(me).await.expect("succeeds").expect("is present");
+
+    assert_eq!(own_identity.user_id(), me);
+    assert!(own_identity.is_verified());
+}
+
+#[ignore = "Verification fails?"]
+#[async_test]
+async fn test_login_with_cross_signing_bootstrapping_already_bootstrapped() {
+    // Even if we enabled cross-signing bootstrap for another device, it won't
+    // restart the procedure.
+    let (builder, server) = test_client_builder().await;
+
+    Mock::given(method("POST"))
+        .and(path("/_matrix/client/r0/login"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(&*test_json::LOGIN))
+        .mount(&server)
+        .await;
+
+    Mock::given(method("POST"))
+        .and(path("/_matrix/client/r0/keys/query"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(json!({
+            "device_keys": {
+                "@cheeky_monkey:matrix.org": {
+                    "JERTCKWUWG":{
+                        "algorithms":[
+                           "m.olm.v1.curve25519-aes-sha2",
+                           "m.megolm.v1.aes-sha2"
+                        ],
+                        "device_id":"JERTCKWUWG",
+                        "keys":{
+                           "curve25519:JERTCKWUWG":"XJixbpnfIk+RqcK5T6moqVY9d9Q1veR8WjjSlNiQNT0",
+                           "ed25519:JERTCKWUWG":"48f3WQAMGwYLBg5M5qUhqnEVA8yeibjZpPsShoWMFT8"
+                        },
+                        "signatures":{
+                           "@cheeky_monkey:matrix.org":{
+                              "ed25519:JERTCKWUWG":"Wc67XYem4IKCpshcslQ6ketCE5otubpX+Bh01OB8ghLxl1d6exlZsgaRA57N8RJ0EMvbeTWCweHXXC/UeeQ4DQ",
+                              "ed25519:uXOM0Xlfts9SGysk/yNr0Vn9rgv1Ifh3R8oPhtic4BM":"dto9VPhhJbNw62j8NQyjnwukMd1NtYnDYSoUOzD5dABq1u2Kt/ZdthcTO42HyxG/3/hZdno8XPfJ47l1ZxuXBA"
+                           }
+                        },
+                        "user_id":"@cheeky_monkey:matrix.org"
+                    }
+                }
+            }
+        })))
+        .mount(&server)
+        .await;
+
+    let client = builder
+        .with_encryption_settings(matrix_sdk::encryption::EncryptionSettings {
+            auto_enable_cross_signing: true,
+        })
+        .request_config(RequestConfig::new().disable_retry())
+        .build()
+        .await
+        .unwrap();
+
+    let auth = client.matrix_auth();
+    // TODO fails: "Verification equation was not satisfied", in
```

Comment:
Well, does the `/keys/query` response up above contain valid data? Was it correctly signed? Did you modify any of the fields?

Reply:
Yeah I changed a field, I should have figured it'd change the signature :facepalm: I'll make the username match the one in the example from which I copied.